### PR TITLE
Create script cleanup

### PR DESCRIPTION
**Pull request by howardjohn**
Clean up cluster creation

    * Add stuff to tmp folder that is in gitignore
    * Clean up some shellcheck exceptions
    * Set project correctly in some places

    Also, remove the google-cloud-key downloading. This might break
    something -- not clear why we need it. I added that as a seperate commit
    so can easily be reverted. If it is needed, we should comment why.

Note: this is not just bikeshedding, I accidentally leaked my google-cloud-key. Most of this is mitigation for preventing that.

### CODE REVIEW

File: perf/istio-install/create_cluster.sh [hunk 1]
```diff
@@ -262,17 +250,15 @@ data:
   gkeClusterUrl: https://container.googleapis.com/v1/projects/${PROJECT_ID}/locations/${ZONE}/clusters/${CLUSTER_NAME}
 EOF
 
-export KUBECONFIG="${CLUSTER_NAME}/kube.yaml"
+export KUBECONFIG="${WD}/tmp/${CLUSTER_NAME}/kube.yaml"
```
Comment (mandarjog) [inline perf/istio-install/create_cluster.sh]: How about creat a Cluster directory variable and then use it everywhere.

File: perf/istio-install/create_sa.sh [hunk 2]
```diff
@@ -56,7 +56,3 @@ done
 gc projects add-iam-policy-binding "${PROJECT_ID}" --role "roles/meshconfig.writer" --member "serviceAccount:${GCP_CTL_SA}@${PROJECT_ID}.iam.gserviceaccount.com"
 # Required for creating the NEG objects in GCP
 gc projects add-iam-policy-binding "${PROJECT_ID}" --role "roles/compute.admin" --member "serviceAccount:${GCP_CTL_SA}@${PROJECT_ID}.iam.gserviceaccount.com"
-
-if [[ "${CLUSTER_NAME}" != "" ]]; then
```
Comment (mandarjog) [inline perf/istio-install/create_sa.sh]: @costinm removing this will break istiod right.

Comment (howardjohn) [reply perf/istio-install/create_sa.sh]: Why does istiod require this? Seems like it shouldn't, unless we are doing something weird for test purposes? Just running this script triggers an email alerting me I downloaded a SA key.

Happy to add it back if we have a reason though

**Comment by howardjohn** [comment]
@mandarjog I added a var, and added back the secret -- confirmed it is needed. I added a step to delete the file after though

**Comment by stewartbutler** [comment]
@howardjohn @mandarjog I'm going to be submitting a PR by EOD that incorporates the spirit but not the letter of these changes in the process of cleaning up some other issues (e.g. the new changes will make the script idempotent, so it won't die if the service accounts already exist). If there is no urgent need for this right now, we can wait until I am ready to submit that PR and it'll cover most of the problems here.

**Comment by stewartbutler** [comment]
@howardjohn @mandarjog See https://github.com/istio/tools/pull/621

**Pull request closed without merging**